pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -883,6 +892,113 @@ func (r *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 			}
 		}
 	}
+	if allSecrets.regSecret != nil || allSecrets.s3Secret != nil || allSecrets.weaveSecret != nil || allSecrets.vsphereSecret != nil || allSecrets.vcenterSecret != nil || allSecrets.openStackSecret != nil || allSecrets.aadClientSecret != nil || allSecrets.aadCertSecret != nil {
+		if r.ClusterClient == nil {
+			return nil, fmt.Errorf("Error updating the cluster: k8s client is nil")
+		}
+		cluster, err := r.ClusterClient.Get(apiContext.Request.Context(), existingCluster["id"].(string), metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		if allSecrets.regSecret != nil {
+			if _, ok := existingCluster[registrySecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", registrySecretKey)
+			}
+		}
+		if allSecrets.s3Secret != nil {
+			if _, ok := existingCluster[s3SecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", s3SecretKey)
+			}
+		}
+		if allSecrets.weaveSecret != nil {
+			if _, ok := existingCluster[weaveSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", weaveSecretKey)
+			}
+		}
+		if allSecrets.vsphereSecret != nil {
+			if _, ok := existingCluster[vsphereSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", vsphereSecretKey)
+			}
+		}
+		if allSecrets.vcenterSecret != nil {
+			if _, ok := existingCluster[virtualCenterSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", virtualCenterSecretKey)
+			}
+		}
+		if allSecrets.openStackSecret != nil {
+			if _, ok := existingCluster[openStackSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", openStackSecretKey)
+			}
+		}
+		if allSecrets.aadClientSecret != nil {
+			if _, ok := existingCluster[aadClientSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", aadClientSecretKey)
+			}
+		}
+		if allSecrets.aadCertSecret != nil {
+			if _, ok := existingCluster[aadClientCertSecretKey]; ok {
+				values.RemoveValue(cluster.Object, "status", aadClientCertSecretKey)
+			}
+		}
+		_, err = r.ClusterClient.Update(apiContext.Request.Context(), cluster, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	}
+	owner := metav1.OwnerReference{
+		APIVersion: "management.cattle.io/v3",
+		Kind:       "Cluster",
+		Name:       data["id"].(string),
+		UID:        k8sTypes.UID(data["uuid"].(string)),
+	}
+	if allSecrets.regSecret != nil {
+		err = r.secretMigrator.UpdateSecretOwnerReference(allSecrets.regSecret, owner)
+		if err != nil {
+			logrus.Errorf("cluster store: failed to set %s %s as secret owner", owner.Kind, owner.Name)
```

Comment:
Should these error be unique for each secret? Like specifying if it was reg, s3, weave, etc.. Otherwise, we can just use a var/const for the error message.